ports/mimxrt/Makefile

```diff
@@ -152,7 +152,7 @@ ifeq ($(MICROPY_HW_SDRAM_AVAIL),1)
 SRC_HAL_IMX_C += $(MCU_DIR)/drivers/fsl_semc.c
 endif
 
-ifeq ($(MICROPY_PY_MACHINE_SDCARD),1)
+ifeq (1, $(filter 1, $(MICROPY_PY_MACHINE_SDCARD) $(MICROPY_PY_NETWORK_CYW43)))
```

Comment:
Yes, done.

Reply:
The respective MICROPY_PY_MACHINE_SDCARD settings in the mpconfigboard.h files are then obsolete.

Reply:
Yes, it should be moved to boardconfig files.

Reply:
Maybe also renamed to `MICROPY_HW_ENABLE_SDCARD` to be consistent with other ports.

Reply:
> Yes, it should be moved to boardconfig files

It is ATM in the boardconfig files. But this PR moves it to mpconfigport.h.

Reply:
Yes it's currently in `mpconfigboard.mk`s, I was saying it should be in `mpconfigboard.h`, and renamed to `MICROPY_HW_ENABLE_SDCARD` to be consistent with boards.

Reply:
But it's not needed as a Makefile option anymore because the USDHC is compiled per series, so if moved to mpconfigboard.h files, it will work, see the last commit, if you have a different idea we can drop it.

Reply:
> because the USDHC is compiled per series,

I've seen that. And because of that we can remove the individual setting in each board file set and just use one setting in Makefile or mpconfigport.h.

Reply:
Oh okay, I see what you mean, so it will be enabled by default unless defined in `mpconfigboard.h` ?

Reply:
About. I had expected to set  `MICROPY_PY_MACHINE_SDCARD` after the line 
`ifeq ($(MCU_SERIES),$(filter $(MCU_SERIES), MIMXRT1021 MIMXRT1052 MIMXRT1062 MIMXRT1064 MIMXRT1176))`
in Makefile. But the actual state is fine. Didn't you want to rename it to `MICROPY_HW_ENABLE_SDCARD`?

Reply:
> Didn't you want to rename it to `MICROPY_HW_ENABLE_SDCARD`?

I did, but I think it's now more consistent with the other `MICROPY_PY_MACHINE_xxx` options in `mpconfigport.h`. There are 3 ports that have `machine_sdcard.c`, when it becomes standard, I think the correct option to enable it would be  `MICROPY_PY_MACHINE_SDCARD`, the `MICROPY_HW_ENABLE_SDCARD` should be for just enabling `sdcard.c` (for example for C-level usage, sdio.c etc..).